sqsconsumerconfig.go

```diff
@@ -62,3 +62,62 @@ func (c *DefaultSQSQueueConsumerComponent) New(ctx context.Context, config *Defa
 		Queue:    q,
 	}, nil
 }
+
+// SmartSQSQueueConsumerConfig represents the configuration to configure SmartSQSQueueConsumer
+type SmartSQSQueueConsumerConfig struct {
+	AWSEndpoint     string
+	QueueURL        string
+	QueueRegion     string
+	Logger          *log.Config
+	NumWorkers      uint64
+	MessagePoolSize uint64
+}
+
+// Name of the configuration
+func (*SmartSQSQueueConsumerConfig) Name() string {
+	return "sqsworker"
+}
+
+// SmartSQSQueueConsumerComponent enables creating configured Component
+type SmartSQSQueueConsumerComponent struct {
+	Logger          *log.Component
+	NumWorkers      uint64
+	MessagePoolSize uint64
+}
+
+// NewSmartSQSQueueConsumerComponent generates a new SmartSQSQueueConsumerComponent
+func NewSmartSQSQueueConsumerComponent() *SmartSQSQueueConsumerComponent {
+	return &SmartSQSQueueConsumerComponent{
+		Logger: log.NewComponent(),
+	}
+}
+
+// Settings generates the default configuration for DefaultSQSQueueConsumerComponent
+func (c *SmartSQSQueueConsumerComponent) Settings() *SmartSQSQueueConsumerConfig {
+	return &SmartSQSQueueConsumerConfig{
+		Logger: c.Logger.Settings(),
+	}
+}
+
+// New creates a configured SmartSQSConsumer
+func (c *SmartSQSQueueConsumerComponent) New(ctx context.Context, config *SmartSQSQueueConsumerConfig) (SmartSQSConsumer, error) {
+	var sesh = session.Must(session.NewSession())
+	q := sqs.New(sesh, &aws.Config{
+		Region:     aws.String(config.QueueRegion),
+		HTTPClient: http.DefaultClient,
+		Endpoint:   aws.String(config.AWSEndpoint),
+	})
+
+	logger, err := c.Logger.New(ctx, config.Logger)
```

Comment:
task: unfortunately, in our services, we can't create new loggers, because a root logger is already configured by our HTTP runtime, which we use even in our services that solely processes queue-based workloads, because they still need an HTTP-based healthcheck. 

Moreover, because any given application can only have one root logger and stater, components like this one should support the pattern of sharing that root logger, even for teams that don't build services exactly like ours.

So then, instead of making a settings-compatible Logger here, can you convert the `Logger` attribute of the `SmartSQSConsumer` to a `LogFn` that knows how to retrieve a `Logger` from the `context.Context`, similar to how to we build our other components that need to emit logs or stats without instantiating their own root loggers or staters?

Reply:
sure